core/templates/dev/head/pages/state_editor/StateResponsesDirective.js

```diff
@@ -301,10 +301,12 @@ oppia.directive('stateResponses', [
                   var feedbackContentId = GenerateContentIdService.getNextId(
                     COMPONENT_NAME_FEEDBACK);
 
-                  $scope.tmpOutcome = OutcomeObjectFactory.createNew(
-                    stateName, feedbackContentId, '', []);
-                  if ($scope.isInQuestionMode) {
-                    $scope.tmpOutcome.dest = null;
+                  if ($scope.questionModeEnabled) {
```

Comment:
```
$scope.tmpOutcome = OutcomeObjectFactory.createNew(
  $scope.questionModeEnabled ? null : stateName,
  feedbackContentId, '', []);
```

Or declare a new temporary var for the first arg. But in general minimize the branching and code duplication.